data/src/main/java/com/microsoft/azure/kusto/data/http/HttpClientFactory.java

```diff
@@ -1,108 +1,71 @@
-                        SSLConnectionSocketFactory.getDefaultHostnameVerifier()));
-            } catch (NoSuchAlgorithmException e) {
-                LOGGER.error("Failed to set supported protocols", e);
-            }
-        }
+        options.setMaximumConnectionPoolSize(properties.maxConnectionTotal());
+        options.setConnectionIdleTimeout(Duration.ofSeconds(properties.maxIdleTime()));
+        options.setResponseTimeout(properties.timeout());
+        options.setHttpClientProvider(properties.provider());
 
-        return httpClientBuilder.build();
-    }
-
-    /**
-     * A custom connection keep-alive strategy that uses the server instructions set in the {@code Keep-Alive}
-     * response header; if the response doesn't contain a {@code Keep-Alive} header, the client will use a configurable
-     * keep-alive period.
-     */
-    static class CustomConnectionKeepAliveStrategy implements ConnectionKeepAliveStrategy {
+        // Set Keep-Alive headers if they were requested.
+        // NOTE: Servers are not obligated to honor client requested Keep-Alive values
+        if (properties.isKeepAlive()) {
+            Header keepAlive = new Header(HttpHeaderName.CONNECTION.getCaseSensitiveName(), "Keep-Alive");
+            // Keep-Alive is Non-standard from the client so core does not have an enum for it
+            Header keepAliveTimeout = new Header("Keep-Alive", "timeout=" + properties.maxKeepAliveTime());
 
-        private final int defaultKeepAlive;
+            List<Header> headers = new ArrayList<>();
+            headers.add(keepAlive);
+            headers.add(keepAliveTimeout);
 
-        /**
-         * The default keep-alive time.
-         *
-         * @param defaultKeepAlive the keep-alive time expressed in seconds
-         */
-        CustomConnectionKeepAliveStrategy(int defaultKeepAlive) {
-            this.defaultKeepAlive = defaultKeepAlive;
+            options.setHeaders(headers);
         }
 
-        @Override
-        public long getKeepAliveDuration(HttpResponse httpResponse, HttpContext httpContext) {
-            // honor 'keep-alive' header
-            HeaderElementIterator it = new BasicHeaderElementIterator(httpResponse.headerIterator(HTTP.CONN_KEEP_ALIVE));
-            while (it.hasNext()) {
-                HeaderElement he = it.nextElement();
-                String param = he.getName();
-                String value = he.getValue();
-                if (value != null && param.equalsIgnoreCase("timeout")) {
-                    return Long.parseLong(value) * 1000;
-                }
-            }
-            // otherwise keep alive for default seconds
-            return defaultKeepAlive * 1000L;
+        if (properties.getProxy() != null) {
+            options.setProxyOptions(properties.getProxy());
         }
+
+        // Todo: Is the per route connection maximum needed anymore?
```

Comment:
I think some of the new options are missing - the routeplanner and supported protocols, can you make sure it is paired or at least specify all of the changes?